core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_utils/DistanceAverage.java

```diff
@@ -0,0 +1,19 @@
+package fr.sncf.osrd.envelope_utils;
+
+/** This class can be used to compute average values over distances,
+ * adding segments iteratively. */
+public class DistanceAverage {
+    private double totalValue = 0;
+    private double totalLength = 0;
+
+    /** Add a segment of size `length`, with an average value of `value` */
+    public void addSegment(double length, double value) {
+        totalValue += value * length;
```

Comment:
This naming would make more sense to me
```suggestion
        totalWeightedValue += value * length;
```